sympy/ntheory/generate.py

```diff
@@ -222,7 +244,31 @@ def prime(nth):
     n = as_int(nth)
     if n < 1:
         raise ValueError("nth must be a positive integer; prime(1) == 2")
-    return sieve[n]
+    prime_arr = [2,3,5,7,11,13,17]
+    if n<=6:
+        return prime_arr[n-1]
+        
+    from sympy.functions.special.error_functions import li
+    from sympy.functions.elementary.exponential import log
+
+    a = 2 # Lower bound for binary search
```

Comment:
Here, and in the binary search you could use ``m`` instead of ``a``. Neither is it necessary to use ``N``. ``m`` could stand for it as well.